mathesar/views/api.py

```diff
@@ -85,9 +89,19 @@ class RecordViewSet(viewsets.ViewSet):
 
     def list(self, request, table_pk=None):
         paginator = TableLimitOffsetPagination()
-        records = paginator.paginate_queryset(self.queryset, request, table_pk)
-        serializer = RecordSerializer(records, many=True)
-        return paginator.get_paginated_response(serializer.data)
+        filter_form = RecordListFilterForm(request.GET)
+        if filter_form.is_valid():
+            try:
+                records = paginator.paginate_queryset(
+                    self.queryset, request, table_pk,
+                    filters=filter_form.cleaned_data['filters']
+                )
+            except (FieldNotFound, BadFilterFormat) as e:
```

Comment:
Could you document why you're using these specific errors here and generic ones later?

Reply:
This is so we can catch specific sorting errors (`SortFieldNotFound` and `BadSortFormat`) separately down the line in order to return the error in the appropriate field of the return dictionary. It might make more sense in [the context of the sorting PR](https://github.com/centerofci/mathesar/pull/300/files#diff-ed2d8e28f999e57c7acd4c3e92bd4efd27dc832b90ad625913866af4b41a54d3R102)?